cmake/KokkosCore_config.h.in

```diff
@@ -58,7 +58,6 @@
 
 #cmakedefine KOKKOS_COMPILER_CUDA_VERSION @KOKKOS_COMPILER_CUDA_VERSION@
 
-#cmakedefine KOKKOS_ARCH_NONE
```

Comment:
Please remind me why we had this one.

Reply:
It became unused after https://github.com/kokkos/kokkos/issues/2241.

Reply:
It became unused after #2241.